core/src/lib.rs

```diff
@@ -1,16 +1,45 @@
 use serde::{Deserialize, Serialize};
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct PublicKey {
+    pub key: Vec<u8>,
+    pub key_type: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct DFA {
+    pub fwd: Vec<u8>,
+    pub bwd: Vec<u8>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct RegexInfo {
+    pub parts: Vec<(bool, DFA)>, // (is_public, dfa)
+}
+
 #[derive(Debug, Serialize, Deserialize)]
 pub struct Email {
     pub from_domain: String,
     pub raw_email: Vec<u8>,
-    pub public_key_type: String,
-    pub public_key: Vec<u8>,
+    pub public_key: PublicKey,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct EmailWithRegex {
+    pub email: Email,
+    pub regex_info: RegexInfo,
 }
 
 #[derive(Debug, Serialize, Deserialize)]
-pub struct DKIMOutput {
+pub struct EmailVerifierOutput {
```

Comment:
Structs like these can be imported from zkemail.rs (replace other such structs too)